### ISSUE TITLE
Feedly error: the given data was not valid JSON

### ISSUE THREAD

**Issue by TypicalJanner** [Ranchero-Software/NetNewsWire#2493]
For the past few days syncing with Feedly on both iOS and Mac has been problematic.

The first sign of a problem was when syncing would result in no new articles, even though there were some available.

I removed Feedly and added it again.

The first sync and a couple after this would work as expected.

However, sync would then revert back to pulling in no new articles.

This pattern of behaviour would be repeated if I deleted and re-added Feedly.

I have been experiencing similar problems with lire and Fiery Feeds on iOS, so I do not believe that this is a NNW-specific problem. In the case of lire, the app crashes consistently when attempting to refresh feeds. With Fiery Feeds, the app throws and error that the data is in an incorrect format.

Although this may not be a NNW-specific issue, it was suggested on Twitter that I file a report here and include the logs from Console on Mac.

These appear to indicate that there is indeed an underlying issue with Feedly: 

60984	debug	13:58:13.870134+0100	NetNewsWire	Starting sync 87232794-4F46-4347-9D84-456629172FEF
60984	debug	13:58:13.871556+0100	NetNewsWire	Sending article statuses...
60984	debug	13:58:13.871942+0100	NetNewsWire	Done sending article statuses.
60984	debug	13:58:13.872046+0100	NetNewsWire	Requesting collections.
60984	debug	13:58:14.067568+0100	NetNewsWire	Received collections: (
    "user/27584484-7627-400b-85f3-5855c199a4b8/category/Apple",
    "user/27584484-7627-400b-85f3-5855c199a4b8/category/Drupal",
    "user/27584484-7627-400b-85f3-5855c199a4b8/category/Miscellaneous",
    "user/27584484-7627-400b-85f3-5855c199a4b8/category/News",
    "user/27584484-7627-400b-85f3-5855c199a4b8/category/Tech"
)
60984	debug	13:58:14.067709+0100	NetNewsWire	Ensured 5 folders for 5 collections.
60984	debug	13:58:14.067982+0100	NetNewsWire	Processing 31 feeds.
60984	debug	13:58:18.893378+0100	NetNewsWire	Reached end of stream for <private>
60984	debug	13:58:20.342175+0100	NetNewsWire	0 articles updated since last successful sync start date.
60984	debug	13:58:20.582809+0100	NetNewsWire	Requesting 242 articles.
60984	debug	13:58:21.675762+0100	NetNewsWire	Unable to get entries: Error Domain=NSCocoaErrorDomain Code=4864 "The given data was not valid JSON." UserInfo={NSDebugDescription=The given data was not valid JSON., NSCodingPath=(
), NSUnderlyingError=0x600003492820 {Error Domain=NSCocoaErrorDomain Code=3840 "Unable to convert hex escape sequence (no high character) to UTF8-encoded character." UserInfo={NSDebugDescription=Unable to convert hex escape sequence (no high character) to UTF8-encoded character.}}}.
60984	debug	13:58:21.675877+0100	NetNewsWire	Account.FeedlyGetEntriesOperation failed with error: Error Domain=NSCocoaErrorDomain Code=4864 "The given data was not valid JSON." UserInfo={NSDebugDescription=The given data was not valid JSON., NSCodingPath=(
), NSUnderlyingError=0x600003492820 {Error Domain=NSCocoaErrorDomain Code=3840 "Unable to convert hex escape sequence (no high character) to UTF8-encoded character." UserInfo={NSDebugDescription=Unable to convert hex escape sequence (no high character) to UTF8-encoded character.}}}.
60984	debug	13:58:21.676440+0100	NetNewsWire	Sync completed: 87232794-4F46-4347-9D84-456629172FEF
60984	debug	13:58:21.677593+0100	NetNewsWire	Sync took 7.810 seconds

**Comment by brentsimmons** [comment]
This is definitely a Feedly issue. I recommend reporting it to the Feedly folks.

**Comment by vincode-io** [comment]
@TypicalJanner I've been thinking about this and it is probably a specific feed that Feedly isn't handling well.  If Feedly isn't responsive, you could try exporting your OPML file as a back up and then deleting feeds one at a time until you are able to sync without the JSON error.  Then either undo the deletes or reimport your OPML file to get back to where you started from and delete the offending feed.

**Comment by TypicalJanner** [comment]
It does indeed appear to be a specific feed that's the root issue.

The following feed works in the Feedly web interface and with other third-party apps connected to the Feedly account, but seems to break the Feedly service on NNW:

https://www.huffingtonpost.co.uk/feeds/index.xml

Before adding this feed to my Feedly account, NNW works just fine.

However, as soon as that feed is added, I start getting the console messages included in my previous post to this thread. From here on, NNW will not update any feeds.

If I remove the feed (either from NNW or the Feedly web interface), NNW still throws the same console messages and will not update feeds.

The only way to *flush* this problem is to remove Feedly completely from NNW and then add it back.

According to Feedly they have nearly 900 people subscribed to this feed, so it's not a completely obscure feed.

**Comment by vincode-io** [comment]
I tracked this down to a specific article.  https://www.huffingtonpost.co.uk/entry/baby-first-halloween-costumes_uk_5f7da789c5b6e5aba0d2d2c6?guccounter=1

The issue is in the "Baby Boxer" section.  If you read the paragraph associated with it you will note the "trophy" emoji.  Feedly sometimes encodes this as `\uD83C\uDFC6` (correct) and sometimes as `\uDFC6` (incorrect).  This invalid hex escape sequence causes our JSON parser to fail to interpret server response and just gives up.  I was also unsuccessful at parsing it in Python, so this isn't just an Apple platforms bug.  The server result is truly invalid.

I did try to test in Reeder, but Reeder either gives up at that entry or it doesn't sync back that far.  Either way, I can't get that article to show up in Reeder.  I suspect Reeder hits the same problem, but is failing more gracefully than us.

@TypicalJanner if you want to point the Feedly support team to this entry, I've attached a server response body that has the invalid JSON.  Hopefully this will help them track the problem down and get the entry corrected.

[entries.json.zip](https://github.com/Ranchero-Software/NetNewsWire/files/5394236/entries.json.zip)
